src/librustc/mir/mod.rs

```diff
@@ -1924,11 +1921,13 @@ pub enum PlaceBase<'tcx> {
 pub struct Static<'tcx> {
     pub def_id: DefId,
     pub ty: Ty<'tcx>,
+    pub promoted: Option<Promoted>,
```

Comment:
Why not have an `enum StaticKind` between `DefId` and `Promoted`?

Reply:
@oli-obk Any ideas?

Reply:
I was thinking that we could unify even more things if we stopped treating promoteds as special beyond fetching the correct MIR. So basically during MIR-fetching we'd use the `promoted` field, but other than that, all the const eval code around statics and promoteds could be unified.

Reply:
Although... assuming we'll never need to refer to promoteds outside their own function.. Yea differentiating via an enum makes sense.

Reply:
So does this mean the Static struct will now have 4 fields?

Reply:
A promoted is an anonymous static. So e.g.

```rust
let x: &'static i32 = &42;
```

is legal Rust code, because the `42` is put into an anonymous static.

> So does this mean Static will now have 4 fields? The StaticKind Enum will have 2 possible values: Promoted and Static?

The `Static` type will have 2 fields:

```rust
pub struct Static<'tcx> {
    kind: StaticKind,
    ty: Ty<'tcx>,
}
```

and the enum will have two variants:

```rust
pub enum StaticKind {
    Promoted(Promoted),
    Static(DefId),
}
```

This means you won't need the new `DefId` argument in `promote_consts.rs` anymore.

Reply:
Cool! Got it. I’ll make the necessary changes.